File: code/js/src/services/channels/models/PrivateInviteOutputModel.ts
import { Privacy } from '../../../domain/channels/Privacy';
import { Status } from '../../../domain/channels/Status';
import { UserInfo } from '../../../domain/users/UserInfo';

export type PrivateInviteOutputModel = {
  codHash: string;
  privacy: Privacy;
  status: Status;
  userInfo: UserInfo;
  channelId: number;
  channelName: string;
};

export type PrivateInviteOutput = {
  value: PrivateInviteOutputModel;
};

File: code/jvm/src/main/kotlin/pt/isel/daw/channels/http/model/channel/PrivateInviteOutputModel.kt
package pt.isel.daw.channels.http.model.channel

import pt.isel.daw.channels.domain.channels.Privacy
import pt.isel.daw.channels.domain.channels.Status
import pt.isel.daw.channels.domain.user.UserInfo

data class PrivateInviteOutputModel(
    val codHash : String,
    val privacy: Privacy,
    val status: Status,
    val userInfo: UserInfo,
    val channelId: Int,
    val channelName: String
)
